### DISCUSSION TITLE
"None" option at the top or bottom of question?

### DISCUSSION THREAD

**Discussion by SachaG** [Devographics/surveys#210] [General Survey Design]
Currently the "none" or "not applicable" option is at the top of a question because the idea is that if a respondent knows the question doesn't apply to them, they can click it to disable all options and effectively skip the question altogether. 

But there's also an argument to be made that the "none" option should come at the end, as a respondent can't know if none of the options apply to them until they've read through all options. One issue in that case is that it conflicts with the "Show More…" button, since the "none" option would be hidden unless respondents click that button. 

Or, a third method would be to let it switch places on a question-per-question basis. But then that hurts consistency. 

Thoughts? 

cc @ShaineRosewel @LeaVerou

**Comment by LeaVerou** [comment]
First, internal consistency is very important, we don't want to have None at the top on some questions and at the bottom in others. It's important that users can learn where it is so they can select it without thought.
Second, "None" at the bottom is more externally consistent (i.e. with other surveys).
The point about pagination is a good one, but I cannot think of a single use case where the "Show more…" button is actually a good idea. It biases the results towards the displayed choices, and requires an extra click for the participant to actually use it. IMO it should be removed / never used and we should solve the problems it was designed to solve (reducing scrolling for overly long questions) in other ways.

**Reply by ShaineRosewel** [reply]
If this were in the ideal case, it will be a filter question type where user is first asked if it is a yes or a no. And then, depending on the response,another question will be asked - this could be about the specific options.

For questions that have options, it is not advisable to have a long list of options but sometimes, can't be avoided - similar to demographics questions such as religion. I agree with the suggestion to use drop down with scroll if this is not avoidable for the reason of making sure that everything are given the same exposure. 

May not be conclusive but I observed that the volunteers sometimes select None first but deselect it once they see the choices; I am not sure if others will do the same or just leave the options behind after choosing None. 

Things that could happen: 

1) If it's on top, respondent might select it right away assuming he/she completely understood the question - w/o checking the options. The downside is compared to the ideal case which offers yes or no first, this offers a lot of options for yes and one for no. And it is easier to say no, so it is an easy choice.  I am thinking this is okay if the questions are clear enough to convey what it means - so clear that even if i don't see the options, I still have that clear understanding of what it meant.
2) If it's at the bottom and we have `show more` button, we might be getting a lot of skipped questions when they mean `None`, instead of selecting `None`.

**Reply by SachaG** [reply]
> If this were in the ideal case, it will be a filter question type where user is first asked if it is a yes or a no. And then, depending on the response,another question will be asked - this could be about the specific options.

But then that requires one more step, which is more work for the respondent. So I don't know if that's a good solution either?

**Reply by LeaVerou** [reply]
I agree, not sure it would be ideal at all to turn it into a 2 step process.

**Reply by ShaineRosewel** [reply]
Not ideal in our case. Just trying to think how to give it a feel like we are filtering too.

**Comment by ShaineRosewel** [comment]
The idea from one volunteer makes sense, people read from top to bottom. Hence, when you read past `none` and see the other options and realise there is really `none` that you use. It will take extra effort to scroll back to the top to select none again.

**Comment by SachaG** [comment]
I think putting in at the end makes sense, but the reason why I had it in the beginning was to provide a way for users to skip a question straight away. 

If we put it in the end, should we then have a different “skip” or “not applicable” button higher up before the list of options? And if so should that button just do the same thing as selecting “none”, or be stored differently in the data model?

I also think there’s a subtle difference between “none” as in “none of these options apply to me [but I do use option XYZ]” and “none” as in “I don’t use anything in that category”. Hopefully we can make that clear through wording.

**Reply by LeaVerou** [reply]
The problem is that sometimes they know whether they want to skip it from the beginning, whereas other times it depends on the available options. Ideally we shouldn't have questions with so many options that we need actual UI to skip them. The only one I can think of is Industry Sector, and that's last in its section.
